huaweicloud/services/evs/resource_huaweicloud_evs_snapshot.go

```diff
@@ -121,12 +129,19 @@ func resourceEvsSnapshotV2Read(_ context.Context, d *schema.ResourceData, meta i
 
 	log.Printf("[DEBUG] Retrieved volume %s: %+v", d.Id(), v)
 
+	md := make(map[string]string)
+
+	for key, val := range v.Metadata {
+		md[key] = val
+	}
+
 	mErr := multierror.Append(nil,
 		d.Set("volume_id", v.VolumeID),
 		d.Set("name", v.Name),
 		d.Set("description", v.Description),
 		d.Set("status", v.Status),
 		d.Set("size", v.Size),
+		d.Set("metadata", md),
```

Comment:
To avoid triggering changes metadata is not backfilled during read.
Please refer to `metadata` in resource huaweicloud_bms_instance